gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/messages/httpget/HttpGetEntrypointMockEndpointIntegrationTest.java

```diff
@@ -0,0 +1,316 @@
+
+        final TestSubscriber<JsonObject> obs = createGetRequest("/test", MediaType.APPLICATION_JSON, httpClient)
+            .doOnSuccess(response -> assertThat(response.statusCode()).isEqualTo(200))
+            .flatMapPublisher(response -> response.rxBody().flatMapPublisher(this::extractMessages))
+            .test()
+            .awaitDone(30, TimeUnit.SECONDS)
+            .assertValueCount(messageCount)
+            .assertComplete();
+
+        verifyMessagesAreOrdered(messageCount, obs);
+        verifyMessageHeadersAndMetadata(messageCount, obs);
+    }
+
+    @Test
+    @DeployApi("/apis/v4/messages/http-get/http-get-entrypoint-mock-endpoint.json")
+    void should_receive_messages_xml(HttpClient httpClient) {
+        final int messageCount = 12;
+
+        final TestSubscriber<JsonObject> obs = createGetRequest("/test", MediaType.APPLICATION_XML, httpClient)
+            .doOnSuccess(response -> assertThat(response.statusCode()).isEqualTo(200))
+            .flatMapPublisher(response -> response.rxBody().flatMapPublisher(this::extractXmlMessages))
+            .test()
+            .awaitDone(30, TimeUnit.SECONDS)
+            .assertValueCount(messageCount)
+            .assertComplete();
+
+        verifyMessagesAreOrdered(messageCount, obs);
+    }
+
+    @Test
+    @DeployApi("/apis/v4/messages/http-get/http-get-entrypoint-mock-endpoint.json")
+    void should_receive_messages_text_plain(HttpClient httpClient) {
+        final int messageCount = 12;
+
+        final TestSubscriber<JsonObject> obs = createGetRequest("/test", MediaType.TEXT_PLAIN, httpClient)
+            .doOnSuccess(response -> assertThat(response.statusCode()).isEqualTo(200))
+            .flatMapPublisher(response -> response.rxBody().flatMapPublisher(this::extractPlainTextMessages))
+            .test()
+            .awaitDone(30, TimeUnit.SECONDS)
+            .assertValueCount(messageCount)
+            .assertComplete();
+
+        verifyMessagesAreOrdered(messageCount, obs);
+    }
+
+    @Test
+    @DeployApi("/apis/v4/messages/http-get/http-get-entrypoint-mock-endpoint-timeout.json")
+    void should_return_504_when_gateway_timeout_is_reached(HttpClient httpClient) {
+        final int messageCount = 0;
+
+        final TestObserver<JsonObject> obs = createGetRequest("/test", MediaType.APPLICATION_JSON, httpClient)
+            .doOnSuccess(response -> assertThat(response.statusCode()).isEqualTo(504))
+            .flatMap(response -> response.rxBody().map(body -> new JsonObject(body.toString())))
+            .test()
+            .awaitDone(30, TimeUnit.SECONDS)
+            .assertValue(body -> {
+                assertThat(body.getString("message")).isEqualTo("Request timeout");
+                assertThat(body.getInteger("http_status_code")).isEqualTo(504);
+                return true;
+            })
+            .assertComplete();
+    }
+
+    @NonNull
+    private Flowable<JsonObject> extractMessages(Buffer body) {
```

Comment:
```suggestion
    private Flowable<JsonObject> extractJsonMessages(Buffer body) {
```

Reply:
I just wanted to keep the same name I used on other tests by default.